### ISSUE TITLE
[Bug] when audio focus is regained, symphony auto resumes even when the playback is manually paused

### ISSUE THREAD

**Issue by Icarus-Xu** [zyrouge/symphony#201]
### Steps to Reproduce

1. enable "ignore audio focus loss"
2. play some music in symphony
3. pause music, switch to some streaming app (bilibili in my case)
4. play some video
5. pause video
6. symphony resumes

### Expected Behavior

symphony shall stay paused.

### Screenshots

I ruined that scene by mistake, still trying to reproduce (not easy), will attach asap.

### Device

-   Device: Mi Mix / Redmi K30 Pro
-   OS: LineageOS 18.1 / LineageOS 20.
-   Version [e.g. 2023.6.93]

### Additional Context

So I did some digging and I think the problem is here:

```kotlin
class RadioFocus(val symphony: Symphony) {
    private var restoreOnFocusGain = false
    private val audioFocusRequest: AudioFocusRequestCompat =
        AudioFocusRequestCompat.Builder(AudioManagerCompat.AUDIOFOCUS_GAIN)
            ......
            .setOnAudioFocusChangeListener { event ->
                when (event) {
                    AudioManager.AUDIOFOCUS_GAIN -> {
                        if (restoreOnFocusGain) {
                            restoreOnFocusGain = false
                            when {
                                symphony.radio.isPlaying -> symphony.radio.restoreVolume()
                                else -> symphony.radio.resume()
                            }
                        }
                    }
                    AudioManager.AUDIOFOCUS_LOSS, AudioManager.AUDIOFOCUS_LOSS_TRANSIENT -> {
                        restoreOnFocusGain = true
                        if (!symphony.settings.ignoreAudioFocusLoss.value) {
                            symphony.radio.pause()
                        }
                    }
                    AudioManager.AUDIOFOCUS_LOSS_TRANSIENT_CAN_DUCK -> {
                        restoreOnFocusGain = true
                        if (symphony.radio.isPlaying) {
                            symphony.radio.duck()
                        }
                    }
                }
            }
            .build()
    ......
}
```

When the bug appears, that is when I pause the video from other app, I believe symphony received audio focus because I saw following log:

```text
2023-06-23 15:55:19.682 11941-11941 AudioManager            io.github.zyrouge.symphony.debug     D  dispatching onAudioFocusChange(1) to android.media.AudioManager@c280ab9io.github.zyrouge.symphony.services.radio.RadioFocus$$ExternalSyntheticLambda0@e9ba9a0

```

So symphony, while still in the background (I was tapping pause button in other app), received audio focus, and because `restoreOnFocusGain` is set to `true` when symphony losses audio focus (when the video is played), then symphony proceeded to either restore volume or resume playback, without checking previous status.

I'm not sure why symphony re-gains audio focus in this case, actually in my test it does not happen every time, still trying to find a pattern, but it COULD be an android behaviour, which we may not able to change, so I guess our solution may be check player status in this case, if the player was playing before it losses focus, then we do resume, and we stay calm otherwise.

### Checklist

-   [x] I have read the [Contribution Guidelines](https://github.com/zyrouge/symphony/wiki/Contributions-Guidelines#issues).

**Comment by Icarus-Xu** [comment]
https://github.com/zyrouge/symphony/assets/11155311/a6248a4e-175d-4a59-952f-feaddb95023d

**Comment by Icarus-Xu** [comment]
I think this bug is probably the same with #132

**Comment by zyrouge** [comment]
Damn, thanks for the PR, was thinking of the same solution. My dumbass forgot to check if the player was not paused.

**Comment by zyrouge** [comment]
:tada: Fixed as of v2023.7.104!
